### PR TITLE
Add new functionality to animals table and demonstrate transactions and SQL queries

### PR DESCRIPTION
**Pull request by Oscar-Espinoza**
This PR adds new functionality to the animals table in the database. Here are the changes made:

- Added a new column "species" to the animals table.
- Inserted new animal records into the animals table.
- Demonstrated transactions by updating the species column of all animals to "unspecified", rolling back the transaction, and verifying that the species column went back to its original state. Also updated the species column of certain animals to "digimon" or "pokemon", committed the transaction, and verified that the changes were made and persisted.
- Demonstrated transactions by deleting all records from the animals table, rolling back the transaction, and verifying that all records still exist. Also deleted certain animals based on their birthdate and updated their weight, rolled back to a savepoint, updated their weight again, committed the transaction, and verified that the changes were made and persisted.
- Created SQL queries to answer questions about the animals table, such as the number of animals, the number of animals that have never tried to escape, the average weight of animals, and the minimum/maximum weight of each animal type.

Actions screenshots: 

action_1
action_2
action_3
action_4

Queries screenshots:

q_1
q_2

Overall, this PR adds new functionality to the animals table and demonstrates various concepts such as transactions and SQL queries.

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -6,5 +6,6 @@ CREATE TABLE animals (
   date_of_birth DATE,
   escape_attempts INTEGER,
   neutered BOOLEAN,
-  weight_kg DECIMAL(5,2)
-);
\ No newline at end of file
+  weight_kg DECIMAL(5,2),
+  species VARCHAR(100)
```
Comment (mikethreels) [inline schema.sql]: - [ ] when working on a project it is not possible to change an existing table, you should change the table using the `ALTER TABLE` command.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Oscar, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Felix45 — CHANGES_REQUESTED** [review]
Hi @Oscar-Espinoza , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request :red_circle: 

You have submitted a project, that has a readme file that is not customized to the project. Please customize your readme file and request another review.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by anagudelogu — APPROVED** [review]
# STATUS: APPROVED! 🍾 🎆 🥇 

Hi @Oscar-Espinoza, 
I'm Andres.

I'm Approving your project to be consistent with the previous reviewer, however, having professional documentation is required for every project (it is a general requirement) and not meeting it makes the review invalid. I strongly suggest you update your README, and it is in fact present in the requirements: 🤓 🔧 

image

[Professional documentation link](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).

image

You did a great job working on this milestone, your project is complete! 🚀 
There is nothing else to say other than... it's time to merge it 🤖  
Congratulations! 🎉 



## Highlights
✅ Great use of Git flow. 💯 
✅ Descriptive PR title, summary and commit messages. 💬 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**